datastore/google/cloud/datastore/batch.py

```diff
@@ -133,7 +135,8 @@ def _add_delete_key_pb(self):
         :returns: The newly created key protobuf that will be
                   deleted when sent with a commit.
         """
-        new_mutation = self.mutations.add()
+        new_mutation = _datastore_pb2.Mutation()
+        self._mutations.append(new_mutation)
         return new_mutation.delete
 
     @property
```

Comment:
@daspecster I am :+1: to that but this interface is "beta" so I didn't want to break it.

I'm 99% sure that the `mutations` property was added by me to aid in the transition from `v1beta3` to `v1`.

@lukesneeringer WDYT?